esphome/components/sml/text_sensor/sml_text_sensor.cpp

```diff
@@ -0,0 +1,58 @@
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+#include "sml_text_sensor.h"
+#include "../sml_parser.h"
+
+namespace esphome {
+namespace sml {
+
+static const char *const TAG = "sml_text_sensor";
+
+SmlTextSensor::SmlTextSensor(const char *server_id, const char *obis, const char *format) {
+  this->server_id = std::string(server_id);
+  this->obis = std::string(obis);
+  this->format_ = std::string(format);
+}
```

Comment:
This probably needs formatting as I wrote it on GitHub =)
```suggestion
SmlTextSensor::SmlTextSensor(std::string server_id, std::string obis, std::string format) : server_id(std::move(server_id)), obis(std::move(obis)), format_(std::move(format)) {}
```